src/asn1acn.cpp

```diff
@@ -174,6 +177,16 @@ void Asn1AcnPlugin::initializeMenus()
     contextMenu->addAction(command);
     asnToolsMenu->addAction(command);
 
+
+    asnToolsMenu->addSeparator();
+    QAction *importFromAsnComponents = new QAction(tr("Import from ASN.1 components library..."), this);
+    command = Core::ActionManager::registerAction(importFromAsnComponents, Constants::IMPORT_FROM_COMPONENTS_LIBRARY);
+    connect(importFromAsnComponents, &QAction::triggered, this, &Asn1AcnPlugin::raiseImportComponentWindow);
+    asnToolsMenu->addAction(command);
+    importFromAsnComponents->setEnabled(ProjectExplorer::SessionManager::hasProjects());
+    connect(ProjectExplorer::ProjectExplorerPlugin::instance(), &ProjectExplorer::ProjectExplorerPlugin::fileListChanged,
+            [importFromAsnComponents]() { importFromAsnComponents->setEnabled(ProjectExplorer::SessionManager::hasProjects()); });
```

Comment:
I think it's a good moment to do some refactoring in this method - it got awfully long, we could at least split it some like "createImportAsnComponentsAction" etc.

Reply:
Splitted the function into smaller chunks.